graylog2-server/src/main/java/org/graylog/plugins/views/search/events/MessagesExportEvent.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.search.events;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.graylog.plugins.views.search.export.ExportMessagesCommand;
+import org.graylog2.plugin.indexer.searches.timeranges.AbsoluteRange;
+import org.joda.time.DateTime;
+
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.graylog.plugins.views.audit.ViewsAuditEventTypes.MESSAGES_EXPORT_REQUESTED;
+import static org.graylog.plugins.views.audit.ViewsAuditEventTypes.MESSAGES_EXPORT_SUCCEEDED;
+
+@AutoValue
+@JsonAutoDetect
+public abstract class MessagesExportEvent {
+
+    public static MessagesExportEvent requested(DateTime startTime, String userName, ExportMessagesCommand command) {
+        return from(startTime, userName, command, MESSAGES_EXPORT_REQUESTED);
+    }
+
+    public static MessagesExportEvent succeeded(DateTime startTime, String userName, ExportMessagesCommand command) {
+        return from(startTime, userName, command, MESSAGES_EXPORT_SUCCEEDED);
+    }
+
+    private static MessagesExportEvent from(DateTime startTime, String userName, ExportMessagesCommand command, String auditType) {
+        return Builder.create()
+                .userName(userName)
+                .auditType(auditType)
+                .timestamp(startTime)
+                .timeRange(command.timeRange())
+                .queryString(command.queryString().queryString())
+                .streams(command.streams())
+                .fieldsInOrder(command.fieldsInOrder())
+                .build();
+    }
+
+    public abstract String userName();
+
+    public abstract String auditType();
+
+    public abstract DateTime timestamp();
+
+    public abstract AbsoluteRange timeRange();
+
+    public abstract String queryString();
+
+    public abstract Set<String> streams();
```

Comment:
I would add the `limit` for a rough estimation how many messages were exported. Or perhaps the total message count in the succeed case?